### DISCUSSION TITLE
How can you silence Liesel's "info" messages?

### DISCUSSION THREAD

**Discussion by Seb-Lorek** [liesel-devs/liesel#161] [Q&A] [answered]
In general I really like the print statements (from `logging`) during model building and inference  of `liesel` , especially when using `liesel` in a `.ipynb` file or in a "small" application. But I'm searching for a way to silence Liesel's "info" messages such that the console is not cluttered with all the print statements when running `liesel` in a python script. This would be helpful when using `liesel` in a larger project, where I might execute different subtasks and want to print only "global" information on the different subtasks rather than the "infos"/"warnings" of `liesel`. I saw that there is the option to set 

```python
engine._show_progress = False
```

This prohibits the print statement of the sampling progress. But I'm currently not aware on how to disable the other print statements (or catch them). Happy to receive some input on that.

**Comment by jobrachem** [comment, marked as answer]
Hi @Seb-Lorek, really nice to see you regularly contributing here 👍 

Since we use Python's `logging` module, I think your desired behavior can be achieved by adjusting the log level. 

Consider the following example: 

![image](https://github.com/liesel-devs/liesel/assets/37882800/4fd530a4-0433-41ba-aac8-8d41936e9c86)


There are five different logging levels. Most of the verbose feedback from liesel can be silenced by setting the logging level to "warning", as in the example above. You can even control the behavior for the logging messages issued by individual submodules, if you replace the `logger = logging.getLogger("liesel")` above for example with the more specific `logger = logging.getLogger("liesel.model.model")`.  For the sampling messages issued by goose, you might want to set the log level of the `"liesel.goose.engine"` logger to `logging.ERROR`, if you are not interested in those messages. I would be hesitant to recommend setting the log level generally to this value, though, because you might miss something important.

![image](https://github.com/liesel-devs/liesel/assets/37882800/44cad597-a7f9-4a15-bbba-60b756b992e3)

Here's the docs of the `logging` module: https://docs.python.org/3/howto/logging.html#logging-basic-tutorial

**Reply by Seb-Lorek** [reply]
Nice this is exactly what I was looking for, thanks!

**Reply by jeli2103** [reply]
Hello, 

I agree that some messages might be of importance. So you better save them to a file:
```python
# loggers (info level)
logger_liesel = logging.getLogger('liesel')
logger_liesel.setLevel(logging.INFO)

# create file handler which logs to file
fh = logging.FileHandler('spam.log')
fh.setLevel(logging.INFO)

# create formatter and add
formatter = logging.Formatter('%(name)s - %(levelname)s - %(message)s')
fh.setFormatter(formatter)

# add handler
logger_liesel.addHandler(fh)
```

So you have all outputs in the spam.log file. However, I was not able to find a solution to log to a file but not logging to the console (but this should not be a liesel question..., but if you have the answered at hand, help is appreciated)

**Reply by jobrachem** [reply]
@liesel-groupie thanks for the code example! 😊 I think it is ok if we treat your question about logging to a file + silencing output to the console as a liesel question, since it depends a little on liesel's default logging setup. To keep things organized, I would recommend that you open another question for that, then I will provide a solution there.
